lib/SILGen/SILGenDestructor.cpp

```diff
@@ -219,6 +220,135 @@ void SILGenFunction::destroyClassMember(SILLocation cleanupLoc,
   }
 }
 
+llvm::SmallSetVector<VarDecl*, 4> findRecursiveLinks(DeclContext* DC, ClassDecl *cd) {
+  auto SelfTy = DC->mapTypeIntoContext(cd->getDeclaredInterfaceType());
```

Comment:
You don't need the mapTypeIntoContext() here; it will cause the check to fail with generic types since you didn't do mapTypeIntoContext() on the property type below

Reply:
Ah, forgot to remove this. Thanks!